src/Microsoft.IdentityModel.Tokens/Exceptions/SecurityTokenSignatureValidationFailedException.cs

```diff
@@ -0,0 +1,76 @@
+﻿//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+using System.Runtime.Serialization;
+
+namespace Microsoft.IdentityModel.Tokens
+{
+    /// <summary>
+    /// This exception is thrown when a security token does not contain a key identifier and the signature validation is not successful with keys found by the runtime.
+    /// </summary>
+    [Serializable]
+    public class SecurityTokenSignatureValidationFailedException : SecurityTokenSignatureKeyNotFoundException
```

Comment:
We need to be _careful_ with this

It should work with [OIDC](https://github.com/dotnet/aspnetcore/blob/bbf7c8780c42e3d32aeec8018367037784eb9181/src/Security/Authentication/OpenIdConnect/src/OpenIdConnectHandler.cs#L748)

It won't with [WsFed](https://github.com/dotnet/aspnetcore/blob/5ff9ed68d1cf6c89d72d27a69b00ed0ecd34daed/src/Security/Authentication/WsFederation/src/WsFederationHandler.cs#L298)

Reply:
Good point. JwtBearer is coded in a similar way as OIDC (`ex is SecurityTokenSignatureKeyNotFoundException`) which should work fine with the proposed model. 

However, we are not breaking the refresh scenario in WsFed with this change as Wilson is currently throwing `SecurityTokenInvalidSignatureException` when there is no 'kid', and keys couldn't validate the signature.

The right thing to do here would be to create a pull request for WsFed, so it's consistent with other handlers.
We would need similar PRs in Katana:
[oidc](https://github.com/aspnet/AspNetKatana/blob/635c92f641ad1e014eead31cc7a365004949fda5/src/Microsoft.Owin.Security.OpenIdConnect/OpenidConnectAuthenticationHandler.cs#L526)
[wsfed](https://github.com/aspnet/AspNetKatana/blob/e2b18ec84ceab7ffa29d80d89429c9988ab40144/src/Microsoft.Owin.Security.WsFederation/WsFederationAuthenticationHandler.cs#L372)

Reply:
That is a Good point, for WsFed it wouldn't break the refresh scenario but #1268 will not be addressed. 

I agree about creating pull request for other handlers.

That makes me think of another scenario where the user is also using similar method as WsFed ( `exception.GetType().Equals(typeof(SecurityTokenSignatureKeyNotFoundException` ) to check what exception it is and is performing some action based on it, that would break things ? as we are currently throwing a derived exception( `SecurityTokenSignatureValidationFailedException` ) in place of `SecurityTokenInvalidSignatureException` .

Reply:
@keegan-caruso nice catch. I opened https://github.com/dotnet/aspnetcore/issues/24847 against asp.net.

@RojaEnnam  We can not make this change to a new exception. We will have stay with the existing exception.